volume/drivers/pwx/connection.go

```diff
@@ -132,27 +140,74 @@ func (cpb *ConnectionParamsBuilder) BuildClientsEndpoints() (string, string, err
 		} else if svcPort.Name == cpb.Config.DefaultRestPortName &&
 			svcPort.Port != 0 {
 			restPort = int(svcPort.Port)
+		} else if svcPort.Name == cpb.Config.DefaultRestPortNameSecured &&
+			svcPort.Port != 0 {
+			restPortSecured = int(svcPort.Port)
 		}
 	}
+	// if secured REST port (9023) is available, use it instead of legacy 9001
+	if restPortSecured != 0 {
+		finalRestPort = restPortSecured
+	} else {
+		finalRestPort = restPort
+	}
 
 	// check if the ports were parsed
-	if sdkPort == 0 || restPort == 0 {
-		err := fmt.Errorf("%s in %s namespace does not contain %s and %s ports set", serviceName, ns, cpb.Config.DefaultSDKPortName, cpb.Config.DefaultRestPortName)
+	if sdkPort == 0 || finalRestPort == 0 {
+		err := fmt.Errorf("%s in %s namespace does not contain %s and either of %s or %s ports set", serviceName, ns, cpb.Config.DefaultSDKPortName, cpb.Config.DefaultRestPortName, cpb.Config.DefaultRestPortNameSecured)
 		logrus.Errorf(err.Error())
 		return "", "", err
 	}
 
-	// PX mgmt API was decided not to be protected with TLS
-	// Using names not IP, so no need for ipv6 protection here
-	pxMgmtEndpoint = fmt.Sprintf("http://%s:%d", endpoint, restPort)
-	sdkEndpoint = fmt.Sprintf("%s:%d", endpoint, sdkPort)
+	scheme := "http"
+	var isTLSEnabled = isTLSEnabled(cpb.Config.EnableTLSEnv)
+	if isTLSEnabled && finalRestPort == restPortSecured {
+		scheme = "https" // legacy 9001 port is never TLS secured, irrespective of the value of PX_ENABLE_TLS
+	}
+	pxMgmtEndpoint = fmt.Sprintf("%s://%s", scheme, net.JoinHostPort(endpoint, strconv.Itoa(finalRestPort)))
+	sdkEndpoint = net.JoinHostPort(endpoint, strconv.Itoa(sdkPort))
 
 	logrus.Infof("Using %s as endpoint for portworx REST API", pxMgmtEndpoint)
 	logrus.Infof("Using %s as endpoint for portworx gRPC API", sdkEndpoint)
 
 	return pxMgmtEndpoint, sdkEndpoint, nil
 }
 
+// BuildTlsConfig returns the TLS configuration (if needed) to connect to the Porx API
+func (cpb *ConnectionParamsBuilder) BuildTlsConfig() (*tls.Config, error) {
+	var isTLSEnabled = isTLSEnabled(cpb.Config.EnableTLSEnv)
+	if !isTLSEnabled {
```

Comment:
nit: can just be isTLSEnabled(cpb.Config.EnableTLSEnv)
```suggestion
	if isTLSEnabled(cpb.Config.EnableTLSEnv) {
```